src/components/table-hoc/TableWithFilter.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import {keys} from 'ts-transformer-keys/index';
+import * as _ from 'underscore';
+import {IReactVaporState} from '../../ReactVapor';
+import {callIfDefined} from '../../utils/FalsyValuesUtils';
+import {ReduxConnect} from '../../utils/ReduxUtils';
+import {FilterBoxConnected} from '../filterBox/FilterBoxConnected';
+import {IFilterState} from '../filterBox/FilterBoxReducers';
+import {ITableHOCOwnProps} from './TableHOC';
+
+export interface ITableWithFilterConfig {
+    matchFilter?: (filterValue: string, datum: any) => boolean;
+}
+
+export interface ITableWithFilterStateProps {
+    filter: string;
+}
+
+export interface ITableWithFilterDispatchProps {
+    onRender: () => void;
+}
+
+export interface ITableWithFilterProps extends Partial<ITableWithFilterStateProps>,
+    Partial<ITableWithFilterDispatchProps>,
+    ITableHOCOwnProps {}
+
+const TableWithFilterPropsToOmit = keys<ITableWithFilterStateProps>();
+
+const defaultMatchFilter = (filter: string, datum: any) => JSON.stringify(_.values(datum).map((v: string) => v.toLowerCase())).indexOf(filter.toLowerCase()) !== -1;
+
+export const tableWithFilter = (config: ITableWithFilterConfig = {matchFilter: defaultMatchFilter}) => (Component: (React.ComponentClass<ITableWithFilterProps> | React.StatelessComponent<ITableWithFilterProps>)): React.ComponentClass<ITableWithFilterProps & React.HTMLAttributes<HTMLTableElement>> => {
+    const mapStateToProps = (state: IReactVaporState, ownProps: ITableWithFilterProps): ITableWithFilterStateProps | ITableHOCOwnProps => {
+        const filter: IFilterState = _.findWhere(state.filters, {id: ownProps.id});
+        const matchingData = (data: any[]) => {
+            if (filter && filter.filterText) {
+                return _.filter(ownProps.data, (datum: any) => config.matchFilter(filter.filterText, datum));
+            }
+            return data;
+        };
+        return {
+            filter: filter && filter.filterText,
+            data: ownProps.data && matchingData(ownProps.data),
+        };
+    };
+
+    @ReduxConnect(mapStateToProps)
+    class TableWithFilter extends React.Component<ITableWithFilterProps> {
+
+        componentDidUpdate(prevProps: ITableWithFilterProps) {
+            if (prevProps.filter !== this.props.filter) {
+                callIfDefined(this.props.onUpdate);
+            }
+        }
+
+        render() {
+            const filterAction = <FilterBoxConnected
+                key='FilterBox'
+                id={this.props.id}
+                className='coveo-table-actions'
+                isAutoFocus
+            />;
+            const newActions = [...[], ...this.props.actions, filterAction];

```

Comment:
i don't think the first `...[]` is necessary , array spread compiles to Array.concat which returns a new array

Reply:
But this.props.actions can be undefined and in this case the code output doesn't work

![image](https://user-images.githubusercontent.com/260007/47729352-5f4c9800-dc36-11e8-81b7-ef31948d83c7.png)
[playground](http://www.typescriptlang.org/play/#src=interface%20IProps%20%7B%0D%0A%20%20%20%20maybeArr%3F%3A%20any%5B%5D%3B%0D%0A%7D%0D%0Aconst%20props%3A%20IProps%20%3D%20%7B%7D%3B%0D%0A%0D%0A%5B...props.maybeArr%2C%20%22a%22%5D%3B%0D%0A%0D%0A%0D%0A)

Reply:
You're right on that, but in the other case if it's undefined you'll insert an undefined value in the array:

<img width="917" alt="screen shot 2018-10-30 at 2 16 19 pm" src="https://user-images.githubusercontent.com/9539763/47740596-8911b900-dc4e-11e8-9182-1aea34f712c6.png">

i guess something like `[...(this.props.actions || []), filterAction]` should be used